staging/src/k8s.io/dynamic-resource-allocation/structured/namedresources/cel/compile.go

```diff
@@ -1,243 +0,0 @@
-limitations under the License.
-*/
-
-package cel
-
-import (
-	"context"
-	"fmt"
-	"reflect"
-	"sync"
-
-	"github.com/blang/semver/v4"
-	"github.com/google/cel-go/cel"
-	"github.com/google/cel-go/common/types"
-	"github.com/google/cel-go/common/types/traits"
-
-	resourceapi "k8s.io/api/resource/v1alpha2"
-	"k8s.io/apimachinery/pkg/util/version"
-	celconfig "k8s.io/apiserver/pkg/apis/cel"
-	apiservercel "k8s.io/apiserver/pkg/cel"
-	"k8s.io/apiserver/pkg/cel/environment"
-)
-
-const (
-	attributesVarPrefix = "attributes."
-)
-
-var (
-	lazyCompilerInit sync.Once
-	lazyCompiler     *compiler
-)
-
-func GetCompiler() *compiler {
-	lazyCompilerInit.Do(func() {
-		lazyCompiler = newCompiler()
-	})
-	return lazyCompiler
-}
-
-// CompilationResult represents a compiled expression.
-type CompilationResult struct {
-	Program     cel.Program
-	Error       *apiservercel.Error
-	Expression  string
-	OutputType  *cel.Type
-	Environment *cel.Env
-}
-
-type compiler struct {
-	envset *environment.EnvSet
-}
-
-func newCompiler() *compiler {
-	return &compiler{envset: mustBuildEnv()}
-}
-
-// CompileCELExpression returns a compiled CEL expression. It evaluates to bool.
-func (c compiler) CompileCELExpression(expression string, envType environment.Type) CompilationResult {
-	resultError := func(errorString string, errType apiservercel.ErrorType) CompilationResult {
-		return CompilationResult{
-			Error: &apiservercel.Error{
-				Type:   errType,
-				Detail: errorString,
-			},
-			Expression: expression,
-		}
-	}
-
-	env, err := c.envset.Env(envType)
-	if err != nil {
-		return resultError(fmt.Sprintf("unexpected error loading CEL environment: %v", err), apiservercel.ErrorTypeInternal)
-	}
-
-	ast, issues := env.Compile(expression)
-	if issues != nil {
-		return resultError("compilation failed: "+issues.String(), apiservercel.ErrorTypeInvalid)
-	}
-	expectedReturnType := cel.BoolType
-	if ast.OutputType() != expectedReturnType {
-		return resultError(fmt.Sprintf("must evaluate to %v", expectedReturnType.String()), apiservercel.ErrorTypeInvalid)
-	}
-	_, err = cel.AstToCheckedExpr(ast)
-	if err != nil {
-		// should be impossible since env.Compile returned no issues
-		return resultError("unexpected compilation error: "+err.Error(), apiservercel.ErrorTypeInternal)
-	}
-	prog, err := env.Program(ast,
-		cel.InterruptCheckFrequency(celconfig.CheckFrequency),
-	)
-	if err != nil {
-		return resultError("program instantiation failed: "+err.Error(), apiservercel.ErrorTypeInternal)
-	}
-	return CompilationResult{
-		Program:     prog,
-		Expression:  expression,
-		OutputType:  ast.OutputType(),
-		Environment: env,
-	}
-}
-
-var valueTypes = map[string]struct {
-	celType *cel.Type
-	// get returns nil if the attribute doesn't have the type, otherwise
-	// the value of that type.
-	get func(attr resourceapi.NamedResourcesAttribute) (any, error)
-}{
-	"quantity": {apiservercel.QuantityType, func(attr resourceapi.NamedResourcesAttribute) (any, error) {
```

Comment:
all types but intslice were removed in the new file... curious if those moved somewhere else or were no longer needed

Reply:
No longer needed. We simplified the attribute types quite a bit.